packages/block-library/src/query/edit/inspector-controls/index.js

```diff
@@ -99,18 +99,13 @@ export default function QueryInspectorControls( props ) {
 		setQuery( updateQuery );
 	};
 	const [ querySearch, setQuerySearch ] = useState( query.search );
-	const onChangeDebounced = useCallback(
-		debounce( () => {
-			if ( query.search !== querySearch ) {
-				setQuery( { search: querySearch } );
-			}
-		}, 250 ),
-		[ querySearch, query.search ]
+	const debouncedQuerySearch = useMemo(
+		() =>
+			debounce( ( newQuerySearch ) => {
+				setQuery( { search: newQuerySearch } );
+			}, 250 ),
+		[ setQuery ]
 	);
```

Comment:
Formatting can be more compact with a _block body_ (no implicit return) in this case:
```suggestion
	const debouncedQuerySearch = useMemo( () => {
		return debounce( ( newQuerySearch ) => {
			setQuery( { search: newQuerySearch } );
		}, 250 );
	}, [ setQuery ] );
```
Of course, now `return` could be omitted too since the effect using the value was removed.

Reply:
That's a good point, and it reads much better. Thanks, @stokesman!